### ISSUE TITLE
Timer behavior

### ISSUE THREAD

**Issue by axblk** [teeworlds/teeworlds#1690]
Right now the timer behavior is rather inconsistent. Some words on the current state:

There are 3 timers: Warmup, Pause, Game starts...

To start or resume a game the Game starts timer is mandatory. It lasts 3 seconds, tees cannot move meanwhile and it only starts when there are enough players for a new game.

When the server starts a new round itself, it uses the Game starts timer. This is (or rather should be) configurable with `sv_warmup`.

Values for `sv_warmup`:
- `-1` use the player ready mode when `sv_player_ready_mode 1` is set
- `0` disabled - only use the Game starts timer
- otherwise use the Warmup timer with the given value, followed by the Game starts timer

The same behavior also applies for the `restart` command.

Issues:
- `sv_warmup` only works once after reload
- There is no possibility to turn off the Game starts timer (#1635)
- When using the Warmup timer it switches to the Game starts timer when reaching 3 seconds.
Actually the Warmup timer displays the tenths of seconds from 5 to 3, the Game starts timer does not... This looks kinda odd. When starting the Warmup timer with a value below 3, it counts until 0 and then the Game starts timer starts from 3 again.
- When unpausing the game the Pause timer counts until 0... followed by the Game starts timer for 3 seconds
- The player ready mode lacks any UI, a default bind or a hint which key to press

So... what do you think would be the best solution for these things?
Switching the timer for the last 3 seconds feels unnecessary and double timers are fairly annoying... but right now the whole thing completely relies on the Game starts timer.

**Comment by oy** [comment]
Tricky one.
Yeah should be consistent. double timers shouldn't be there.

**Comment by themisak** [comment]
#1635 Has to do with it

I think there should be an option to switch Gamestart Timers and Warumup timers completely off!

**Comment by Dune-jr** [comment]
To me, the pesky problem is #1635. The simplest solution to that I can think of is to add an option that controls the game ready timer as such:

- 0 (default): survival only; 3 seconds
- -1: always off
- **n** (**n** > 0): always on, **n** seconds

I think the issue with the double timers could be fixed client-side by making the warmup and the ready timer more uniform?

**Comment by oy** [comment]
These option sound good.

Dunno about a possible client-side change. The 2 timer differ - during warmup timer you can play and during the gamestart one the game is paused.

**Comment by Dune-jr** [comment]
Yeah, but then I'm not sure how to address the double timer confusion issue

**Comment by oy** [comment]
Could replace the numbers with visuals maybe? Like red, yellow, green circle? Or text "Get ready..."(.<-), "Go!".

**Comment by Dune-jr** [comment]
Sounds nice (if it makes it in time), what about the Warmup, does it count down to 3 or to 0 then?

**Comment by oy** [comment]
Right now it stops at 3 and the gamestart timer steps in, to avoid double timers.
